<p>
  This rule checks that the specified child tag does not appear as a direct child of the specified parent.
</p>
<p>
  Assuming a parent/child combination of <code>&lt;head&gt;/&lt;body&gt;</code>:
</p>

<h2>Noncompliant Code Example</h2>
<pre>
&lt;head&gt;
  ...
  &lt;body&gt;  &lt;!-- Noncompliant --&gt;
    ...
  &lt;/body&gt;
&lt;/head&gt;
</pre>

<h2>Compliant Solution</h2>
<pre>
&lt;head&gt;
  ...
&lt;/head&gt;
</pre>
