<p>
  This rule checks that the specified HTML elements are not present.
</p>
<p>
  For a disallowed list of "font,center":
</p>

<h2>Noncompliant Code Example</h2>
<pre>
  &lt;center&gt;&lt;font color="red"&gt;Hello World!&lt;/font&gt;&lt;/center&gt; &lt;!-- Noncompliant --&gt;
</pre>

<h2>Compliant Solution</h2>
<pre>
  &lt;div class="centerRed"&gt;Hello World!&lt;/div&gt;
</pre>
