<p>
Favicons are shown for example in the browser's address bar, bookmark list, or tabs.
They enable users to quickly identify and recognize websites.
</p>

<h2>Noncompliant Code Example</h2>

<pre>
&lt;head&gt;                                                                  &lt;!-- Non-Compliant --&gt;
  &lt;title&gt;...&lt;title&gt;
&lt;/head&gt;
</pre>

<h2>Compliant Solution</h2>

<pre>
&lt;head&gt;                                                                  &lt;!-- Compliant --&gt;
  &lt;title&gt;...&lt;title&gt;
  &lt;link rel="shortcut icon" href="http://example.com/myicon.ico" /&gt;
&lt;/head&gt;
</pre>
