<h2>Why is this an issue?</h2>
<p>Web applications can be made available in multiple languages through the use of internationalization. This allows the server to plug in the correct
version of a piece of text based on the language chosen, but it requires that internationalization messages be used instead of hard-coded text.</p>
<h3>Noncompliant code example</h3>
<pre>
  &lt;form method="post"&gt;
    &lt;label for="username"&gt;Username:&lt;/label&gt;  &lt;!-- Noncompliant --&gt;
    &lt;input type="text" id="username" name="username"&gt;
    &lt;br&gt;
    &lt;label for="password"&gt;Password:&lt;/label&gt;  &lt;!-- Noncompliant --&gt;
    &lt;input type="password" id="password" name="password"&gt;
    &lt;br&gt;
    &lt;input type="submit" name="submit" value="${buttonValue}"&gt;
  &lt;/form&gt;
</pre>
<h3>Compliant solution</h3>
<pre>
  &lt;form method="post"&gt;
    &lt;label for="username"&gt;&lt;fmt:message key="login.label.username" /&gt;:&lt;/label&gt;
    &lt;input type="text" id="username" name="username"&gt;
    &lt;br&gt;
    &lt;label for="password"&gt;&lt;fmt:message key="login.label.password" /&gt;:&lt;/label&gt;
    &lt;input type="password" id="password" name="password"&gt;
    &lt;br&gt;
    &lt;input type="submit" name="submit" value="${buttonValue}"&gt;
  &lt;/form&gt;
</pre>

