<h2>Why is this an issue?</h2>
<p>Use of <code>&lt;meta http-equiv="refresh"&gt;</code> is discouraged by the World Wide Web Consortium (W3C).</p>
<p>If a user clicks the 'Back' button, some browers will go back to the redirecting page, which will prevent the user from actually going back.</p>
<p>To refresh the page, a better alternative is to use Ajax, to refresh only what needs to be refreshed and not the whole page.</p>
<p>To redirect to another page, using the HTTP response status code 301 'Moved Permanently' and 302 'Found' is a better option.</p>
<h3>Noncompliant code example</h3>
<pre>
&lt;head&gt;
  &lt;meta http-equiv="refresh" content="5"&gt;   &lt;!-- Non-Compliant --&gt;
  &lt;meta name="description" content="..."&gt;
&lt;/head&gt;
</pre>
<h3>Compliant solution</h3>
<pre>
&lt;head&gt;
  &lt;meta name="description" content="..."&gt;
&lt;/head&gt;
</pre>

