<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>Heading tags are used by search engines and screen reader softwares to construct an outline of the page.</p>
<p>Starting at <code>&lt;h1&gt;</code> and not skipping any level eases this automatic construction.</p>
<h3>Noncompliant code example</h3>
<pre>
&lt;h2&gt;My Title&lt;/h2&gt;     &lt;!-- Non-Compliant - h1 is not being used --&gt;

&lt;h3&gt;My Sub Title&lt;/h3&gt; &lt;!-- Compliant --&gt;
</pre>
<h3>Compliant solution</h3>
<pre>
&lt;h1&gt;My Title&lt;/h1&gt;     &lt;!-- Compliant --&gt;

&lt;h2&gt;My Sub Title&lt;/h2&gt; &lt;!-- Compliant --&gt;
</pre>

