<h2>Why is this an issue?</h2>
<p>When parsing a script node, the browser treats its contents as plain text, and immediately finishes parsing when it finds the first closing
<code>&lt;/script&gt;</code> character sequence.</p>
<p>As a consequence, nested script nodes are not possible, because all opening <code>&lt;script&gt;</code> tags found along the way are ignored.</p>
<p>Web browsers doesnâ€™t support nested <code>&lt;script&gt;...&lt;/script&gt;</code> elements. But there is no error in such case and browsers just
close the first encountered <code>&lt;script&gt;</code> tag as soon as a closing <code>&lt;/script&gt;</code> tag is found along the way. So there is
a big chance to display something totally unexpected to the end-users.</p>
<h3>Noncompliant code example</h3>
<pre>
&lt;script type="text/template"&gt;
  &lt;div&gt;
    Hello!
  &lt;/div&gt;
  &lt;script type="text/javascript"&gt;  &lt;!-- Noncompliant --&gt;
    alert("Hi!");
  &lt;/script&gt;
&lt;/script&gt;
</pre>
<h3>Compliant solution</h3>
<pre>
&lt;script type="text/javascript"&gt;
  alert("Hi!");
&lt;/script&gt;

&lt;script type="text/template"&gt;
  &lt;div&gt;
    Hello!
  &lt;/div&gt;
&lt;/script&gt;
</pre>

