<p>This rule is deprecated, and will eventually be removed.</p>
<h2>Why is this an issue?</h2>
<p>The <code>ismap</code> attribute in an <code>img</code> tag creates a server-side image map: The browser sends the coordinates of the clicked point
to the server.</p>
<p>For any person who cannot use a mouse, this form of navigation is inaccessible because it is the position of the cursor on the image that
determines the action.</p>
<p>On the other hand, client-side image maps, which use the <code>usemap</code> attribute allow for each clickable area to specify an alternate text,
enabling accessibility for the blind.</p>
<p>Further, in terms of separation of concerns, it is definitely better to leave the task of mapping pixels to links to the client.</p>
<h3>Noncompliant code example</h3>
<pre>
&lt;a href="click_on_world_map.php" target="_self"&gt;
  &lt;img src="world_map.png" ismap&gt;              &lt;!-- Noncompliant --&gt;
&lt;/a&gt;
</pre>
<h3>Compliant solution</h3>
<pre>
&lt;img src="world_map.png" usemap="#world_map"&gt;

&lt;map name="world_map"&gt;
  &lt;area shape="rect" coords="0,0,10,10" href="france.html" alt="France"&gt;
  &lt;area shape="circle" coords="20,20,10" href="spain.html" alt="Spain"&gt;
  &lt;area shape="circle" coords="30,30,8" href="england.html" alt="England"&gt;
  &lt;!-- ... --&gt;
&lt;/map&gt;
</pre>

