<p>The proper use of white space makes a major contribution to code readability.</p>
<p>This rule raises an issue when there is not a space character after the beginning and before the end of each comment (<code>&lt;!-- ...
--&gt;</code>), directive (<code>&lt;%@ ... %&gt;</code>), and expression (<code>&lt;% ... %&gt;</code>).</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;!--Do the thing--&gt;  &lt;!-- Noncompliant; missing space at beginning and end of text--&gt;
&lt;%@page import="java.io.*,java.util.*" %&gt; &lt;!-- Noncompliant; missing space at beginning --&gt;
&lt;% String title = "My Page";%&gt; &lt;!-- Noncompliant; missing space at end --&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;!-- Do the thing --&gt;
&lt;%@ page import="java.io.*,java.util.*" %&gt;
&lt;% String title = "My Page"; %&gt;
</pre>

